cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -405,14 +411,32 @@ Status CopyStridedArray(PyArrayObject* arr, const int64_t length, MemoryPool* po
 }  // namespace
 
 template <typename ArrowType>
-inline Status NumPyConverter::ConvertData(std::shared_ptr<Buffer>* data) {
+inline Status NumPyConverter::PrepareInputData(std::shared_ptr<Buffer>* data) {
   if (is_strided()) {
```

Comment:
I'm not too sure the details of this - is it possible for a boolean array to be strided?

Reply:
All Numpy arrays can be strided, yes.

Reply:
Ok, thanks. This is handled by the `Ndarray1DIndexer` right?